Docs/lambda-annotations-design.md

```diff
@@ -0,0 +1,358 @@
+
+
+### ASP.NET Core Lambda Functions
+
+The alternative for writing .NET Lambda functions for REST APIs is to use .NET�s ASP.NET Core web framework. This has been a popular feature of our .NET experience since the beginning. It allows developers to write REST APIs in a well known and documented framework. The style is idiomatic .NET using .NET attributes and reflection to map HTTP request data into corresponding to method parameters. Some of the features that are most appreciated with this approach are:
+
+* .NET Attributes for mapping route mapping and HTTP request data
+* Injecting middleware into request pipelines 
+    * A piece of code that runs before or after any request in the project
+* Integrated dependency injection
+    * A central location for registering all of the dependencies for the request in the project.
+* Authentication
+    * Define policies and add annotations to request paths that ensure the caller has the permissions for the policies.
+
+
+Here is an example of a ASP.NET Core REST controller doing similar logic to the low level example mention before.
+
+
+```
+[Route("[controller]")]
+public class MathController : ControllerBase
+{
+    [HttpGet("/plus/{x}/{y}")]
+    public int Plus(int x, int y)
+    {
+        return x + y;
+    }
+
+    [HttpGet("/subtract/{x}/{y}")]
+    public int Subtract(int x, int y)
+    {
+        return x - y;
+    }
+}
+```
+
+
+There are a few significant negative side effect for writing REST APIs for Lambda using ASP.NET Core. 
+
+* Significant impact to cold start due to size of the framework and reliance on reflection.
+* The common features of ASP.NET Core for REST APIs are supported in Lambda but a significant number of the features are not and there is no clear indication what is and isn�t supported.
+* The usage of API Gateway is dumbed down because the API Gateway REST API is defined as a single wild card resource path letting ASP.NET Core handling the routing. This takes away a lot of the features of API Gateway like per resource path IAM permissions and memory sizing. 
+* ASP.NET Core can only be used for defining REST APIs. It can not be used for other types of Lambda functions like S3 events.
+
+## Goals for a better experience
+
+To improve the experience of writing Lambda functions we need an experience that is similar to what developers are used to with ASP.NET Core without the negative side effects. The major goals are:
+
+* An experience similar to what .NET developers are used in frameworks like ASP.NET Core
+    * Although it will follow similar patterns we will not reuse the same classes as ASP.NET Core to avoid an implicit contract agreements between the developer and the framework.
```

Comment:
```suggestion
    * Although it will follow similar patterns we will not reuse the same classes as ASP.NET Core to avoid an implicit contract agreement between the developer and the framework.
```

Reply:
Fixed